stdlib/hashtbl.mli

```diff
@@ -19,6 +19,49 @@
 (** Hash tables and hash functions.
 
    Hash tables are hashed association tables, with in-place modification.
+
+   Hash tables are hashed association tables, with in-place modification.
+   They're the bread-and-butter associative structure for imperative code,
+   with fast lookup.
```

Comment:
This statement seems to me a little unfriendly for beginners. 

Maybe something in the line "Since most operations over this data structure are in-place, i.e., mutating, it is more common in imperative code." would work better? 

Also, while I understand what you mean by "fast lookup", it a little ambiguous.